src/ParallelAlgorithms/LinearSolver/ConjugateGradient/ElementActions.hpp

```diff
@@ -255,30 +245,25 @@ struct UpdateOperand {
   using inbox_tags =
       tmpl::list<Tags::ResidualRatioAndHasConverged<OptionsGroup>>;
 
-  template <typename DbTags, typename... InboxTags, typename Metavariables,
-            typename ArrayIndex>
-  static bool is_ready(const db::DataBox<DbTags>& box,
-                       const tuples::TaggedTuple<InboxTags...>& inboxes,
-                       const Parallel::GlobalCache<Metavariables>& /*cache*/,
-                       const ArrayIndex& /*array_index*/) noexcept {
-    const auto& inbox =
-        get<Tags::ResidualRatioAndHasConverged<OptionsGroup>>(inboxes);
-    return inbox.find(db::get<Convergence::Tags::IterationId<OptionsGroup>>(
-               box)) != inbox.end();
-  }
-
   template <typename DbTagsList, typename... InboxTags, typename Metavariables,
             typename ArrayIndex, typename ActionList,
             typename ParallelComponent>
-  static std::tuple<db::DataBox<DbTagsList>&&, bool, size_t> apply(
-      db::DataBox<DbTagsList>& box, tuples::TaggedTuple<InboxTags...>& inboxes,
-      const Parallel::GlobalCache<Metavariables>& /*cache*/,
-      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
-      const ParallelComponent* const /*meta*/) noexcept {
-    auto received_data = std::move(
-        tuples::get<Tags::ResidualRatioAndHasConverged<OptionsGroup>>(inboxes)
-            .extract(db::get<Convergence::Tags::IterationId<OptionsGroup>>(box))
-            .mapped());
+  static std::tuple<db::DataBox<DbTagsList>&&, Parallel::AlgorithmExecution,
+                    size_t>
+  apply(db::DataBox<DbTagsList>& box,
+        tuples::TaggedTuple<InboxTags...>& inboxes,
+        const Parallel::GlobalCache<Metavariables>& /*cache*/,
+        const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+        const ParallelComponent* const /*meta*/) noexcept {
+    auto& inbox =
+        get<Tags::ResidualRatioAndHasConverged<OptionsGroup>>(inboxes);
+    const auto& iteration_id =
+        db::get<Convergence::Tags::IterationId<OptionsGroup>>(box);
+    if (inbox.find(iteration_id) == inbox.end()) {
+      return {std::move(box), Parallel::AlgorithmExecution::NotReady, 0};
```

Comment:
Yes, it is ignored.  (Retrying and also switching action doesn't make sense.)  Requiring it to be max is a good idea.

Reply:
Question (no change request for this PR): What do you think about making the `std::tuple` return value of actions a struct with defaults for its constructor arguments? Then this unused action index could be omitted and some of the Algorithm.hpp compile-time logic simplifies. I remember we discussed that possibility a while back, I think @moxcodes suggested it? To be clear, I'm not asking you to do this :)

Reply:
Yeah, I think a dedicated struct is a good way to go long-term.  It would also be a good opportunity to stop making everyone return `std::move(box)`.  (I mean, we could do that now, but might as well wait until we're rewriting everything anyway.)